packages/fxa-auth-db-mysql/scripts/read-security-events.js

```diff
@@ -0,0 +1,164 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * This script takes the uid of a user and will print a security event object of that user
+ * from various tables like securityEvents, sessionTokens and devices tables from fxa database
+ */
+
+/* Usage:
+ *    using email: node read-security-events.js -e email
+ *    Or
+ *    using uid: node read-security-events.js -u uid
+ */
+
+const log = { trace: console.log, error: console.error, stat: console.log, info: console.log } //eslint-disable-line no-console
+const errors = {
+  invalidFlag: 'Invalid Flag Error: Only use -u flag for uid or -e flag for email',
+  invalidUid: 'Invalid UID Error: Pass a valid uid as an argument',
+  invalidEmail: 'Invalid Email Error: Pass a valid email as an argument',
+  missingFlagOrValue: 'Error: Pass the uid or email after -u or -e flag',
+}
+const DB = require('../lib/db/mysql')(log, require('../db-server').errors)
+const config = require('../config')
+
+const flag = process.argv[2]
+const value = process.argv[3]
+const validFlags = ['-e', '-u']
+
+if (! flag || ! value) {
+  log.error(errors.missingFlagOrValue)
+  process.exit()
+}
+
+if (flag && validFlags.indexOf(flag) === -1) {
+  log.error(errors.invalidFlag)
+  process.exit()
+}
+
+const uid = (flag === '-u') ? value : false
+const email = (flag === '-e') ? value : false
+
+const user = {}
+const securityEventsObject = {}
+
+if (uid) {
+  if (uid.length !== 32) {
+    log.error(errors.invalidUid)
+    process.exit()
+  }
+
+  user.uid = Buffer.from(uid, 'hex')
+  readEvents()
+}
+
+if (email) {
+  if (! email.includes('@')) {
+    log.error(errors.invalidEmail)
+    process.exit()
+  }
+
+  readEvents()
+}
+
+function readEvents() {
+  DB.connect(config).then(async db => {
+    try {
+      if (email) {
+        const uid = await readUidByEmail(db, email)
+        user.uid = Buffer.from(uid, 'hex')
+      }
+
+      const devices = await readDevices(db)
+      securityEventsObject.devices = devices
+
+      const sessions = await readSessionTokens(db)
+      securityEventsObject.sessions = sessions
+
+      const events = await readSecurityEvents(db)
+      securityEventsObject.securityEvents = events
+
+      log.info(securityEventsObject)
+
+      db.close()
+    } catch (err) {
+      log.error(err.stack || err.message || err)
+      process.exit()
+    }
+  })
+}
+
+async function readUidByEmail(db, email) {
+  let { uid } = await db.accountExists(email)
+  uid = uid.toString('hex')
+
+  return uid
+}
+
+async function readDevices(db) {
+  try {
+    const { uid } = user
+    const devices = await db.accountDevices(uid)
+
+    if (devices.length) {
+      for (const d of devices) {
+        d.id = d.id.toString('hex')
+        d.uid = d.uid.toString('hex')
+        d.sessionTokenId = d.sessionTokenId.toString('hex')
+        d.refreshTokenId = d.refreshTokenId.toString('hex')
```

Comment:
d.refreshTokenId is not always guaranteed to exist. I don't think you need to include it.